contracts/LifToken.sol

```diff
@@ -48,18 +48,11 @@ contract LifToken is MintableToken, Pausable {
      @return true if the call function was executed successfully
    */
   function approveData(address spender, uint256 value, bytes data) public whenNotPaused returns (bool) {
-
     require(spender != address(this));
 
-    allowed[tx.origin][spender] = value;
-
-    if (spender.call(data)) {
-      Approval(tx.origin, spender, value);
-      return true;
-    } else {
-      return false;
-    }
-
+    super.approve(spender, value);
+    assert(spender.call(data));
```

Comment:
I think here we should use require instead of assert.

Reply:
Agreed, I just changed it